### DISCUSSION TITLE
Installation of FIDI - Bad Gateway

### DISCUSSION THREAD

**Discussion by benfolds** [firefly-iii/firefly-iii#7063] [Firefly III]
I read al lot about the Bad Gateway error (docs and git), but i seem to have another problem? 
**Firefly III runs great** on this webhosting (LAMP) with nginx. Installing `data-importer`, i get the Bad Gateway Error.
But:
- no error to be found in the laravel error logs (error reporting is on)
- no error in the apache2 logs
- no error in the nginx logs

I (repectively the hoster) changed the nginx settings to:
```
proxy_buffer_size       128k;
proxy_buffers           4 256k;
proxy_busy_buffers_size 256k;
```
but that also did not help. 
Do you have any other idea or can i provide some more information?

**Comment by JC5** [comment]
Without any errors there's not much to debug. Is there an entry in the web server's access.log?

**Comment by benfolds** [comment]
I know, that makes it quite difficulte to handle 

`log/nginx/access.log`

```
217.232.242.228 - - [21/Feb/2023:10:48:15 +0100] "GET https://{DOMAIN}/ HTTP/2.0" 302 358 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"
217.232.242.228 - - [21/Feb/2023:10:48:15 +0100] "GET https://{DOMAIN}/token HTTP/2.0" 502 552 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"
```

I also have some debug and info in `storage/logs/data-import-2023-02-20.log `

```
[2023-02-20 15:06:23] local.DEBUG: No access token in hasAccessToken(), will return config variable.
[2023-02-20 15:06:23] local.DEBUG: No client id in hasClientId(), will return config variable.
[2023-02-20 15:06:23] local.DEBUG: No base url in getBaseUrl(), will return config variable.
[2023-02-20 15:06:23] local.DEBUG: No vanity url in getVanityUrl(), will return config variable.
[2023-02-20 15:06:23] local.INFO: The following configuration information was found:  
[2023-02-20 15:06:23] local.INFO: Personal Access Token: "{token}" (limited to 25 chars if present)
[2023-02-20 15:06:23] local.INFO: Client ID            : "0"
[2023-02-20 15:06:23] local.INFO: Base URL             : "https://{DOMAIN}"
[2023-02-20 15:06:23] local.INFO: Vanity URL           : "https://{DOMAIN}"
[2023-02-20 15:06:23] local.DEBUG: Found personal access token + URL "https://{DOMAIN}" in config, set cookie and return to index.
```

**Reply by JC5** [reply]
> 217.232.242.228 - - [21/Feb/2023:10:48:15 +0100] "GET https://{DOMAIN}/token HTTP/2.0" 502 552 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36"

Alright, so something arrives at Firefly III. These are usually the requests that give a 502 (back) when headers are too long etc. Nothing special in the Firefly III logs?

**Reply by benfolds** [reply]
There are no new entries at `storage/logs` when i call the domain. Are there some other logs for this projekt?
As i wrote above, Firefly III is up and runnig, this is about the `data-importer`.

**Reply by JC5** [reply]
Yes you're right. It's weird. The set where you end up at sets a bunch of cookies and then returns you to the index. I can't tell why it would crash. See https://github.com/firefly-iii/data-importer/blob/main/app/Http/Controllers/TokenController.php#L206

**Comment by benfolds** [comment]
Fixed it! Seems like there was another nginx issue. This helped:

```
server {
...
    fastcgi_buffers  16 16k;
    fastcgi_buffer_size  32k;
}
```

Thank you for your work and support!

**Reply by JC5** [reply]
Thanks for the feedback! I'll make sure to include this in the documentation as well.

**Comment by lorenzopicoli** [comment]
For nginxproxymanager users, a simple
```
  proxy_buffer_size   128k;
  proxy_buffers   4 256k;
  proxy_busy_buffers_size   256k; 
```

under "advanced configuration" of the host proxy will suffice

**Comment by benfolds** [comment]
```php
fastcgi_buffers  16 16k;
fastcgi_buffer_size  32k;
``` 

Fixed it
